AsyncDisplayKit/TextKit/ASFunctor.h

```diff
@@ -0,0 +1,59 @@
+//
+//  ASFunctor.h
+//  AsyncDisplayKit
+//
+//  Copyright (c) 2014-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+//
+
+#import <Foundation/Foundation.h>
+
+/* generic functors */
+
+namespace ASDK {
+  
+  template<class T>
+  struct DescribeFunctor {
+    NSString *operator()(const T &t) const {
+      return [NSString stringWithFormat:@"%d", static_cast<int>(t)];
```

Comment:
We can profile this, but I'm confident this will show up as inefficient.

The rest of this diff is quite beautiful in its use of efficient techniques :).  We should look at some kind of itoa implementation or find the code in the Pinterest app that does this more efficiently (I think Jon Parise wrote it)